wdl/GatherSampleEvidence.wdl

```diff
@@ -377,3 +479,141 @@ task LocalizeReads {
     File output_index = basename(reads_index)
   }
 }
+
+
+task CheckAligner {
+  input {
+    File reads_path
+    File reads_index
+    File? reference_fasta
+    File? reference_index
+    File? reference_dict
+    String sample_id
+
+    String gatk_docker
+    RuntimeAttr? runtime_attr_override
+  }
+
+  parameter_meta {
+    reads_path: {
+                 localization_optional: true
+               }
+  }
+
+  RuntimeAttr default_attr = object {
+                               cpu_cores: 1,
+                               mem_gb: 1.0,
+                               disk_gb: 10,
+                               boot_disk_gb: 10,
+                               preemptible_tries: 3,
+                               max_retries: 1
+                             }
+  RuntimeAttr runtime_attr = select_first([runtime_attr_override, default_attr])
+
+  output {
+    File header = "~{sample_id}.header.sam"
+    Int is_dragen_3_7_8 = read_int("is_dragen_3_7_8.txt")
+  }
+  command <<<
+    set -euo pipefail
+
+    gatk PrintReadsHeader \
+      -I ~{reads_path} \
+      --read-index ~{reads_index} \
+      -O ~{sample_id}.header.sam \
+      ~{"-R " + reference_fasta}
+
+    awk '$0~"@PG" && $0~"ID: DRAGEN SW build" && $0~"VN: 05.021.604.3.7.8"' ~{sample_id}.header.sam \
+      | wc -l \
+      > is_dragen_3_7_8.txt
+  >>>
+  runtime {
+    cpu: select_first([runtime_attr.cpu_cores, default_attr.cpu_cores])
+    memory: select_first([runtime_attr.mem_gb, default_attr.mem_gb]) + " GiB"
+    disks: "local-disk " + select_first([runtime_attr.disk_gb, default_attr.disk_gb]) + " HDD"
+    bootDiskSizeGb: select_first([runtime_attr.boot_disk_gb, default_attr.boot_disk_gb])
+    docker: gatk_docker
+    preemptible: select_first([runtime_attr.preemptible_tries, default_attr.preemptible_tries])
+    maxRetries: select_first([runtime_attr.max_retries, default_attr.max_retries])
+  }
+}
+
+task RealignSoftClippedReads {
```

Comment:
This task is pretty tied to scramble, maybe move it to `Scramble.wdl`?

Reply:
I tried moving it in but it adds a lot of wdl code. One thing that's kind of awkward is having to pass in the scramble_table from the previous step. I think there's less code this way and avoids adding a lot of mutually exclusive inputs to the Scramble wdl itself (all the bwa files etc).